[ONNX] Relax constant_fold gather with indices rank > 1 (#68140)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69550
* #69549
* #69548
* #69547
* #69546
* #69544
* __->__ #68493
* #68492
* #68491
* #68490
* #68489

Fixes #66786.

`index_select` only supports `index` of 1-D tensor. `ONNX::Gather` allows `index` to have rank `q`. Abort constant folding `ONNX::Gather` if `index` rank is larger than 1.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D32483826](https://our.internmc.facebook.com/intern/diff/D32483826)

BowenBao committed:
[ONNX] Relax constant_fold gather with indices rank > 1 (#68140)

Fixes #66786.

`index_select` only supports `index` of 1-D tensor. `ONNX::Gather` allows `index` to have rank `q`. Abort constant folding `ONNX::Gather` if `index` rank is larger than 1.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32483826).

BowenBao committed:
Update on "[ONNX] Relax constant_fold gather with indices rank > 1 (#68140)"


Fixes #66786.

`index_select` only supports `index` of 1-D tensor. `ONNX::Gather` allows `index` to have rank `q`. Abort constant folding `ONNX::Gather` if `index` rank is larger than 1.

Co-authored-by: BowenBao <bowbaomicrosoft.com>

Differential Revision: [D32483826](https://our.internmc.facebook.com/intern/diff/D32483826)

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32483826).

BowenBao committed:
Update on "[ONNX] Relax constant_fold gather with indices rank > 1 (#68140)"


Fixes #66786.

`index_select` only supports `index` of 1-D tensor. `ONNX::Gather` allows `index` to have rank `q`. Abort constant folding `ONNX::Gather` if `index` rank is larger than 1.

Co-authored-by: BowenBao <bowbaomicrosoft.com>

Differential Revision: [D32483826](https://our.internmc.facebook.com/intern/diff/D32483826)

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32483826).
